### DISCUSSION TITLE
YOLOv8 Train error

### DISCUSSION THREAD

**Discussion by pinheiromelobruno** [ultralytics/ultralytics#1151] [Q&A]
Hi, I am getting this error while trying to train on YOLOv8:

Traceback (most recent call last):
  File "/usr/local/bin/yolo", line 33, in <module>
    sys.exit(load_entry_point('ultralytics', 'console_scripts', 'yolo')())
  File "/content/ultralytics/ultralytics/yolo/cfg/__init__.py", line 304, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
  File "/content/ultralytics/ultralytics/yolo/engine/model.py", line 298, in train
    self.trainer = TASK_MAP[self.task][1](overrides=overrides)
KeyError: None

**Anyone knows what this means?**

**Comment by AyushExel** [comment]
Please provide the command to reproduce

**Reply by pinheiromelobruno** [reply]
!yolo train model=YOLOv8n.yaml data=custom_data.yaml epochs=300 imgsz=640 batch = -1 patience=300

custom_data.yaml is a yaml pointing to my dataset

**Reply by pinheiromelobruno** [reply]
Apparently the bug it's not working anymore when I try to use the .yaml models (only .pt)

**Reply by glenn-jocher** [reply]
@pinheiromelobruno it seems there might be an issue with the configuration files or the way the command is structured. Ensure that your `model` and `data` YAML files are correctly formatted and located in the right directory. Also, check that the `batch` parameter is set correctly; it should be an integer value without spaces around the equals sign.

If the problem persists, you might want to verify that your environment is set up correctly and that you're using the latest version of the Ultralytics YOLOv8 repository. If you've recently updated the repository, ensure that there are no breaking changes that could affect your training setup.

For further assistance, you can refer to the [Train documentation](https://docs.ultralytics.com/modes/train) for detailed guidance on training YOLOv8 models. If you continue to encounter issues, please provide more details about your setup and the exact steps you've taken so we can help troubleshoot the problem more effectively. 🛠️
